test/distributed/_composable/test_composability/test_fsdp_tp_dp_composability.py

```diff
@@ -0,0 +1,382 @@
+# Owner(s): ["oncall: distributed"]
+
+import copy
+import functools
+from typing import List, Type
+
+import torch
+import torch.distributed as dist
+import torch.distributed.checkpoint as dcp
+import torch.nn as nn
+
+from torch.distributed._composable import checkpoint, replicate
+from torch.distributed._composable.fsdp import CPUOffloadPolicy
+from torch.distributed._composable.fsdp.fully_shard import fully_shard
+from torch.distributed._tensor import DeviceMesh, init_device_mesh
+from torch.distributed._tensor.debug.comm_mode import CommDebugMode
+from torch.distributed.checkpoint.state_dict import (
+    get_model_state_dict,
+    get_optimizer_state_dict,
+)
+from torch.distributed.tensor.parallel import (
+    ColwiseParallel,
+    parallelize_module,
+    RowwiseParallel,
+)
+from torch.distributed.tensor.parallel.ddp import _pre_dp_module_transform
+
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import (
+    check_sharded_parity,
+    FSDPTest,
+    MLP,
+    MLPStack,
+)
+
+from torch.testing._internal.common_utils import run_tests, skipIfRocm
+
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    MLPModule,
+    ModelArgs,
+    Transformer,
+)
+from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
+
+
+# Tensor-Parallel degree
+TP_DEGREE = 2
+LR = 3e-5
+
+c10d_ops = torch.ops.c10d
+funcol = torch.ops.c10d_functional
+
+
+def init_model(device_type, model_parallel_size=TP_DEGREE):
```

Comment:
Similarly here, these should be at a tighter scope specific to the tests that use it.

Reply:
This function also does not appear to be used in this file.